src/amber/support/params.cr

```diff
@@ -0,0 +1,42 @@
+require "./predicates"
+
+class Check
+  def self.it(match)
+    match.nil? ? false : true
+  end
+end
+
+# Monkey patch the
+class String
+  include Amber::Support::Predicates::String
+end
+
+abstract struct Number
+  include Amber::Support::Predicates::Number
+end
+
+module Amber
+  module Support
+    class Params
+      property params
+      property valid : Bool = false
+
+      def initialize(params : HTTP::Params)
+        @params = params
+      end
+
+      def validate(&block)
+        result = with self yield
+        @valid = result
+      end
+
+      def valid?
```

Comment:
you can set on top of class:
```cr
property? valid = false
```
The `property?` will generate the `valid?` getter for you.